File: Lista4/classes/StringOfPoints2D/StringOfPoints2D.cpp
#include "StringOfPoints2D.h"

/** Create empty vector set of Points 2D */
StringOfPoints2D::StringOfPoints2D() {
    pointsSet = new std::vector<Point2D>;
}

/** Make free space */
StringOfPoints2D::~StringOfPoints2D() {
    delete pointsSet;
}

/** Add new point to set of points */
void StringOfPoints2D::addPoint(Point2D pointToAdd) {

}

/** Remove point from set of points */
void StringOfPoints2D::removePoint(Point2D pointToRemove) {

}

/** Check if point is in set of points */
bool StringOfPoints2D::isPointInStringOfPoints(Point2D pointToCheck) {
    return true;
}

/** Return point with index == 'i' */
Point2D StringOfPoints2D::getPoint(int i) {
    return Point2D(0, 0);
}

/** Return count of all elements in set of points */
int StringOfPoints2D::getStringOfPointsSize() {
    return pointsSet->size();
}

File: Lista5Java/src/StringOfPoints2d.java
import java.util.ArrayList;

/**
 * Class for creating string of 2D points
 */
public class StringOfPoints2d {
    private final ArrayList<Point2d> points;

    public StringOfPoints2d() {
        this.points = new ArrayList<>();
    }

    /**
     * Add new point 2D to array
     * @param pointToAdd new point to add
     */
    public void addPoint(Point2d pointToAdd) {
        this.points.add(pointToAdd);
    }

    /**
     * Remove point from array
     * @param pointToRemove point to remove
     */
    public void removePoint(Point2d pointToRemove) {
        this.points.remove(pointToRemove);
    }

    /**
     * Check if point is in string of points
     * @param pointToCheck point to check
     * @return true if point exists in string
     */
    public boolean isPointInString(Point2d pointToCheck) {
        return this.points.contains(pointToCheck);
    }

    /**
     * Return specific point from string of points
     * @param index index to get
     * @return point from string with index === 'index'
     */
    public Point2d getPoint(int index) {
        return this.points.get(index);
    }

    /**
     * Return size of string of points
     * @return size of string of points
     */
    public int getSize() {
        return this.points.size();
    }

    /**
     * Print all points from string of points
     */
    public void printPoints() {
        for (Point2d point : this.points) {
            point.printPoint();
        }
    }
}
